src/traits.rs

```diff
@@ -10,12 +10,24 @@ use wasm_bindgen::prelude::*;
 use crate::{
     constants::*,
     enums::*,
-    js_collections::JsObjectId,
     local::{ObjectId, Position, RawObjectId, RoomName},
     objects::*,
     pathfinder::SingleRoomCostResult,
+    prelude::*,
 };
 
+pub trait FromReturnCode {
```

Comment:
Why `ReturnCode` now that it does not exist? There should be a comment explaining that this return code is actually the i8 returned by wasm_bindgen calls. Might be a good idea to put `type ReturnCode = i8;` somewhere and use it everywhere in API for clarity.

Reply:
I suppose in my 'naming' thinking I had the mind that even with no more `ReturnCode` type, the concept of a 'return code' is the game's, the set of i8s that map to the specific error values in constants. I dunno that it'd ever be anything other than i8 or if it'd be useful to bind that type?

Reply:
Probably JS API return code will eternally fit in i8, but calling these i8s by `ReturnCode` alias might make the code more readable and it will be easier to search for `ReturnCode` occurrences when refactoring than to search for i8s. In the end, this won't matter much to the average end user as they will use new `Result<..., ErrorCode>` instead.